eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -58,6 +58,7 @@ jobs:
     steps:
       - script: |
           echo "##vso[build.addbuildtag]Scheduled"
+          pip install markdown2
```

Comment:
This isn't going to run all the time, see the condition below. 

The more I see external steps like this being required the more I think we should have one wrapper script step here  that we run in the pipeline instead of inline in maven, or have a wrapper script that we call maven. Otherwise I don't think people will easily be able to figure out how to successfully run these.

Reply:
I believe the only thing we need to do in order to accomplish this is enumerate all the readme.md files under `sdk/${{parameters.ServiceDirectory}}`.  That should be pretty easy to do in your python script, once that is done then we just add one script step:
```
- script: |
     pip install markdown2
     python ./eng/pipelines/scripts/generate_overview_from_readme.py --rf sdk/${{parameters.ServiceDirectory}}
```

Reply:
Good catch on the pip install of markdown2, I've moved that to an unconditional step. 

The purpose of the way I'd done the execution of the script is to produce it for artifact being built before the packaging, on the readme.md that exists in that artifact's directory. This would needlessly process other readme.md files are different levels (Cosmos is an example of this). If you feel strongly I can tear out the python exec calls out of the profile but, unfortunately, the -Dgenerate-overview is still going to be there because of conditional appended configuration to add the overview option to the javadoc command. Also, this is better for testing changes locally since it allows producing the overview as part of the build when that option is set instead of having to run a script on the source tree prior.

Reply:
I'm not personally super worried about a little over processing of readme.md files under a given service folder but we can leave this how it is for now. My biggest concern is around injecting the need for python in the middle of the maven execution and having to know you need a particular version of python already installed on the path as well as a version of markdown2 already installed. Whenever there are extra dependencies like that it becomes hard to track down all the things that are needed to be setup in order to work with the tools.

Reply:
One thing that may mitigate, but not alleviate, your concern is that I ensured my script has no python 3 dependencies and markdown2 is documented to work with 2.6+ or 3.3+ versions. I do agree with the concern about adding things like this into maven execution.